File: src/main/java/pl/com/tegess/controller/admin/readcontroller/AdminReadController.java
package pl.com.tegess.controller.admin.readcontroller;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import pl.com.tegess.controller.admin.request.AdminReadData;
import pl.com.tegess.domain.admin.AdminRepository;

import java.util.List;
import java.util.stream.Collectors;

@RestController
@Component
public class AdminReadController {

    @Autowired
    AdminRepository repository;

    @RequestMapping(value = "/api/admins", method = RequestMethod.GET)
    public List<AdminReadData> get() {
        return repository.find().asList()
                .stream()
                .map(admin -> new AdminReadData(admin.getUsername()))
                .collect(Collectors.toList());
    }

    @RequestMapping(value = "/api/loginAdmin", method = RequestMethod.GET)
    public AdminReadData loginAdmin() {
        return new AdminReadData("Szymek");
    }
}

File: src/main/scala/com/tegess/controller/admin/AdminReadController.scala
package com.tegess.controller.admin

import com.tegess.persistance.service.admin.AdminService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.{RequestMapping, RequestMethod, RestController}

@RestController
@Component
class AdminReadController @Autowired() (adminService: AdminService) {

  import AdminReadController._

  @RequestMapping(value = Array("/api/admins"), method = Array(RequestMethod.GET))
  def getAdmins(): List[AdminReadData] = {
    adminService.findAll().map(admin => AdminReadData(admin.getUsername))
  }

}
object AdminReadController {

  case class AdminReadData(username: String)

}